### PR TITLE
Feature/habit checkpoints

### PR DESCRIPTION
**Pull request by ese-varo**
## Explain why this change is being made.
As a registered user with active habits, I want to see the checkpoints of a specific day of the habit, so that I can see all the checkpoints about the habit.

## Explain how this is accomplished.
Once a habit is created, you could see the list of the days duration of the habit, you could press over each day to add, edit o remove a checkpoint

## How do you manually test this?
  - Given the user is logged in, you can press the link Habits over the navbar, then click on 'Add new habit', fill all the form, and once the habit is created you will be redirected to the details page of the habit
  - Once you see the habit details page if you click on any day of the calendar a card with information about the checkpoint for that day is displayed, if there is an existing checkpoint for that day it will allow you to edit it or remove it, if not, an option for adding one would be available.

### Screenshots
- Create Checkpoint
image
image
image
- Edit
image
image

- Delete
image
image

### CODE REVIEW

File: app/assets/stylesheets/habits.scss [hunk 1]
```diff
@@ -3,8 +3,28 @@
 // You can use Sass (SCSS) here: https://sass-lang.com/
 #calendarContainer,
 #organizerContainer {
-  float: left;
-  margin: 5px;
+  // float: left;
+  // margin: 5px;
```
Comment (OmauryMarmolejo) [inline app/assets/stylesheets/habits.scss]: Please remove these lines if you don't need them

File: app/models/habit.rb [hunk 2]
```diff
@@ -9,4 +12,23 @@ def start_date_format
...
     self.end_date.to_s(:long) 
   end
+
+  def has_date(date)
+    (start_date..end_date).include?(date)
+  end
+
+  def current_day
+    day = days.find_by(date: Date.today)
+    return day ? day : days.first
```
Comment (OmauryMarmolejo) [inline app/models/habit.rb]: You don't need this return

File: app/assets/stylesheets/habits.scss (hunk 1 above)
Comment (ese-varo) [reply app/assets/stylesheets/habits.scss]: thanks, I'll do it

File: app/models/habit.rb (hunk 2 above)
Comment (ese-varo) [reply app/models/habit.rb]: that's true, it's not necessary to be that explicit, thanks

**Review by GregVaz — APPROVED** [review]
:rocket:

File: app/assets/stylesheets/note.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the note controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (jvargas98) [inline app/assets/stylesheets/note.scss]: Is this file needed?

File: app/controllers/checkpoint_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+class CheckpointController < ApplicationController
+  before_action :set_day, only: %i[create update]
+  before_action :set_habit, only: :update
+  before_action :set_checkpoint, only: %i[update destroy]
```
Comment (jvargas98) [inline app/controllers/checkpoint_controller.rb]:
```suggestion
  before_action :authenticate_user!
  before_action :set_day, only: %i[create update]
  before_action :set_habit, only: :update
  before_action :set_checkpoint, only: %i[update destroy]
```

**Review by jvargas98 — COMMENTED** [review]
I left some comments 🚀

**Review by LoginALee — APPROVED** [review]
Great job Alvaro! :taco:

**Review by jvargas98 — APPROVED** [review]
🚀
